src/katgpucbf/utils.py

```diff
@@ -322,3 +323,20 @@ def unix_to_adc(self, timestamp: float) -> float:
     def adc_to_unix(self, samples: float) -> float:
         """Convert an ADC sample count to a UNIX timstamp."""
         return samples / self.adc_sample_rate + self.sync_epoch
+
+
+def gaussian_dtype(bits: int) -> np.dtype:
+    """Get numpy dtype for a Gaussian (complex) integer.
+
+    Parameters
+    ----------
+    bits
+        Number of bits in each real component
+    """
+    assert bits in {4, 8, 16, 32}
+    if bits == 4:
+        # 1-byte, packed with real high, imaginary low. Using void rather
+        # than e.g. uint8 avoids accidentally doing arithmetic on it.
+        return np.dtype("V1")
```

Comment:
Did a shallow dive into `numpy.void` and found [this excellent explanation](https://stackoverflow.com/a/25247533), although I still must ask: What do you mean by [emphasis added]
> Using void rather than e.g. uint8 **avoids accidentally doing arithmetic on it**.

Reply:
Actually that link might be confusing you - it's a different use case for `np.void` (which TBH I wasn't even aware counted as "void"). In this case I want the type to be *opaque*. Consider the following:
```
>>> a = np.array([10, 15, -17], np.int8)
>>> np.abs(a)
array([10, 15, 17], dtype=int8)
```
That makes sense if the bytes in `a` really are integers, but if they're packed 4-bit complex numbers, the result is nonsense, because you're not taking the abs of each component. By using an opaque type, we can ensure that such an operation fails instead of producing nonsense, so that accidentally trying to do things the naive way blows up in your face instead of causing subtle bugs:
```
>>> b = a.view("V1")
>>> b
array([b'\x0A', b'\x0F', b'\xEF'], dtype='|V1')
>>> np.abs(b)
UFuncTypeError: ufunc 'absolute' did not contain a loop with signature matching types <class 'numpy.dtypes.VoidDType'> -> None
```

Reply:
Aha, I see what you mean. Excellent choice.